backend/remote/backend.go

```diff
@@ -255,6 +265,28 @@ func (b *Remote) Configure(obj cty.Value) tfdiags.Diagnostics {
 					`Terraform Enterprise client: %s.`, err,
 			),
 		))
+		return diags
+	}
+
+	// Retrieve the organization entitlements.
+	_, err = b.client.Organizations.Read(context.Background(), b.organization)
```

Comment:
I think this is the wrong layer to do this.

Instead, I think we should include a flag in the _workspace_ object that indicates whether that workspace supports remote operations, and then the server can decide when to set that. Right now, it would just check this organization-level entitlement on the client's behalf, but this leaves our options open for the future to e.g. allow users to turn off remote operations for particular workspaces even if they are entitled to use remote operations.

In general I think this is getting very smelly in terms of how many specific design details are getting baked into the CLI rather than being handled on the server and so I think it'd be worth a round of internal design discussion on how best to architect this rather than just continuing to iteratively pile on complexity here.

Reply:
Good observation! But if you look closely you'll see that we currently don't check anything here at all, other then if the organization exists. So despite of the comment (which I updated partly as a reminder) nothing's changed here with regards to the previous functionality.

As both the new entitlements API and updates to the workspaces API are not done yet, this PR is only meant to change the way (and location in the code path) the remote backend can fall back to the local backend so that we can build on that when the APIs are updated.

Still you made a valid point! Especially given the wording of the comment, I understand (and fully agree with) your reasoning. As a matter of fact we actually already discussed this internally and came to the same conclusion (will share the link in a DM) :wink:

So I think this PR can move forward without any changes, but we should still have a discussion to talk this through to make sure we are on the same page.